crates/bevy_ecs/src/change_detection.rs

```diff
@@ -284,6 +336,26 @@ impl<'a> Ticks<'a> {
     }
 }
 
+/// Shared borrow of an entity's component with change tracking metadata
```

Comment:
Needs links to explain "change tracking metadata" in these docs.

Reply:
There is no such documentation. I can point over to `DetectChangesMut` though.

Reply:
Yeah, that sounds reasonable. We should probably have module level docs here, but that can wait for its own PR.

Reply:
I think it should be rephrased with less jargon. Something like "Shared borrow of an entity's component that gives access to change detection."